### PR TITLE
[JBPM-9580] Stabilize CustomNotificationListenerTest

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-9580](https://issues.redhat.com/browse/JBPM-9580)

### CODE REVIEW

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/CustomNotificationListenerTest.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ public static void generalSetup() {
     public void setup() {
         unit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
         deploymentService.deploy(unit);
+        countDownListener.setSaveContent(false);
```
Comment (afalhambra) [inline kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/CustomNotificationListenerTest.java:85]: Looks good to me, though, not really sure why this bean is being shared among the test cases.

Another different approach I've tried is to use the following inside `AfterClass` or even `After` annotated methods in both test cases:

- https://github.com/kiegroup/droolsjbpm-integration/blob/21879389f74c1c8ea44ae0269262db9c4e58990a/kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/CustomNotificationListenerTest.java
- https://github.com/kiegroup/droolsjbpm-integration/blob/21879389f74c1c8ea44ae0269262db9c4e58990a/kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/DeadlinesNotificationTest.java

`NotificationListenerManager.get().reset()` - By doing this, all notification listeners will be removed from the cache, and thus no one will be reused aftewards.

Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/CustomNotificationListenerTest.java]: Yes, good idea, that's a better approach. Thanks for suggestion @afalhambra !

**Review by afalhambra — APPROVED** [review]
Great job @gmunozfe! Thanks!

**Comment by afalhambra** [comment]
Hi @gmunozfe - I think we should try to merge this one. There were some random issues in CI lately regarding this issue.

**Comment by gmunozfe** [comment]
@mareknovotny could you merge? thanks!

**Comment by mareknovotny** [comment]
shouldn't we back port this to 7.52.x too ? @gmunozfe

**Comment by gmunozfe** [comment]
> shouldn't we back port this to 7.52.x too ? @gmunozfe

@mareknovotny It's already there: https://github.com/kiegroup/droolsjbpm-integration/commit/321fd6d5319908fafa635cdfcf186cd8b9f177cd

**Pull request merged**